src/libmongoc/tests/test-mongoc-topology-scanner.c

```diff
@@ -179,7 +179,7 @@ test_topology_scanner_discovery (void)
 
    uri_str = bson_strdup_printf ("mongodb://%s/?" MONGOC_URI_REPLICASET "=rs",
                                  mock_server_get_host_and_port (primary));
-   client = test_framework_client_new (uri_str, NULL);
+   client = mongoc_client_new (uri_str);
```

Comment:
Please update this file to use test_framework_client_pool_new_from_uri and replace the calls to mock_server_receives_legacy_hello with mock_server_receives_any_hello.